pkg/jwt/forwarding_signature_validator.go

```diff
@@ -0,0 +1,26 @@
+package jwt
+
+import (
+	"sync/atomic"
+)
+
+type forwardingSignatureValidator struct {
```

Comment:
Considering that this struct's public API is greater than SignatureValidator (i.e., there is a Replace() method), this should just be a public type.

Reply:
Done. I would normally have preferred to return the struct, and not the interface, but that seemed not to be the common pattern here, at least not in the jwt package. I've updated this to return the ForwardingSignatureValidator struct type.